### PR TITLE
Run end-to-end tests in Github Actions

### PR DESCRIPTION
**Pull request by bartjkdp**
## Improvements

This PR integrates the Cypress end-to-end tests with Github Actions.

- Tests run in parallel
- Reports are aggregated in the Cypress.io dashboard

## Todo

- [x] Automated provisioning of access token (hardcoded one expires).
- [x] Put Cypress in a separate package.
- [x] Configure `CYPRESS_RECORD_KEY` secret in Amsterdam/signals-frontend
- [x] Add documentation how to run e2e-tests

Closes Signalen/frontend#64

### CODE REVIEW

File: environment.conf.e2e.json [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  },
+  "logo": {
+    "url": null,
+    "width": 0,
+    "height": 0,
+    "smallWidth": 0,
+    "smallHeight": 0
+  },
+  "sentry": null,
+  "matomo": null,
```
Comment (boris-arkenaar) [inline environment.conf.e2e.json:18]: These lines can be removed now.

File: environment.conf.e2e.json [hunk 2]
```diff
@@ -0,0 +1,109 @@
+{
```
Comment (boris-arkenaar) [inline environment.conf.e2e.json:1]: Is this file necessary? It's yet another copy of the whole configuration that needs to be maintained.

File: internals/webpack/webpack.base.babel.js [hunk 3]
```diff
@@ -124,6 +124,7 @@ module.exports = options => ({
       NODE_ENV: JSON.stringify(process.env.NODE_ENV),
       GIT_BRANCH: JSON.stringify(process.env.GIT_BRANCH) || 'dummy',
       BUILD_ENV: JSON.stringify(process.env.BUILD_ENV) || 'development',
+      ENABLE_SERVICEWORKER: JSON.stringify(process.env.ENABLE_SERVICEWORKER) || '1',
```
Comment (boris-arkenaar) [inline internals/webpack/webpack.base.babel.js:127]: Can we use a boolean value here?

Comment (bartjkdp) [reply internals/webpack/webpack.base.babel.js:127]: Probably not, environment variables are always a string.

Comment (boris-arkenaar) [reply internals/webpack/webpack.base.babel.js:127]: You could do something like

```
      ENABLE_SERVICEWORKER: Boolean(JSON.stringify(process.env.ENABLE_SERVICEWORKER)),
```

or

```
      ENABLE_SERVICEWORKER: JSON.stringify(process.env.ENABLE_SERVICEWORKER) === '1',
```

Comment (bartjkdp) [reply internals/webpack/webpack.base.babel.js:127]: That could work.

File: environment.conf.e2e.json (hunk 2 above)
Comment (bartjkdp) [reply environment.conf.e2e.json:1]: I agree, not a fan of this solution as well. When need to merge #1094 in order to be able to inherit the base configuration.

File: internals/webpack/webpack.base.babel.js (hunk 3 above)
Comment (bartjkdp) [reply internals/webpack/webpack.base.babel.js:127]: Hmm thought of this a bit more, I think I am fan of keeping this a string as the result of this plugin is exposed as enviroment variable in Javascript, through `process.env`. I would expect this to be a string, not a boolean.

File: environment.conf.e2e.json (hunk 2 above)
Comment (bartjkdp) [reply environment.conf.e2e.json:1]: I think we can remove this file now, right?

**Review by aditudorache — COMMENTED** [review]
LGTM; I've added a couple of suggestions. A lot of tests are failing at this moment, We can merge this as soon as those are fixed.

File: e2e-tests/cypress.json [hunk 4]
```diff
@@ -1,12 +1,12 @@
 {
   "baseUrl": "http://localhost:3001",
   "defaultCommandTimeOut": 5000,
-  "projectId": "b6phb4",
+  "projectId": "xdzm5k",
```
[the remark this answers is not in the document]
Comment (bartjkdp) [reply e2e-tests/cypress.json:4]: This is a new project on Cypress.io. It is already configured in the Amsterdam/signals-frontend repository.

[the remark this answers is not in the document]
Comment (bartjkdp) [reply e2e-tests/cypress.json:4]: The failing end-to-end tests are not blocking for the workflow. I would suggest merging this in and fixing the tests on the go as they keep breaking.

File: start.sh [hunk 5]
```diff
@@ -5,8 +5,10 @@ set -e
 node /internals/scripts/validate-config.js
 node /internals/scripts/inject-config.js
 
-export SIGNALS_SERVICE_WORKER_VERSION=$(date +"%Y-%m-%d %T")
-envsubst < /usr/share/nginx/html/sw.js > /tmp/sw.js
-mv /tmp/sw.js /usr/share/nginx/html/sw.js
+if [ -f /usr/share/nginx/html/sw.js ]; then
```
[the remark this answers is not in the document]
Comment (bartjkdp) [reply start.sh:9]: Done 👍

**Review by aditudorache — APPROVED** [review]
